cpp/src/gandiva/CMakeLists.txt

```diff
@@ -27,14 +27,18 @@ find_package(LLVM)
 
 # Set the path where the byte-code files will be installed.
 set(GANDIVA_BC_INSTALL_DIR
-  ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/gandiva)
+  ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/gandiva)
 
 set(GANDIVA_BC_FILE_NAME irhelpers.bc)
```

Comment:
Probably need to give this an unambiguous name, like `gandiva_irhelpers.bc`?

Reply:
I've added "gandiva_" prefix.
But I don't think that it's unambiguous because it's installed into `gandiva/` sub directory. I feel that `gandiva/gandiva_irhelpers.bc` is redundant. Should we install this as `lib/gandiva_irhelpers.bc` instead of `lib/gandiva/gandiva_irhelpers.bc`?

Reply:
I prefer https://github.com/apache/arrow/pull/3232/commits/ab60eda9677fc69391f7e5aea15488c40ebdc813

The change installs the file to `lib/gandiva/irhelper.bc` and puts the built file to `${BUILD_DIR}/src/gandiva/irhelpers.bc` (instead of `${BUILD_DIR}/{debug,release}/irhelpers.bc`).

If we don't use `${BUILD_DIR}/{debug,release}/` directory as `irhelpers.bc` output directory, `irhelpers.bc`'s isn't unambiguous. `irhelpers.bc` isn't executable, static library nor shared library. So I think that we don't need to put `irhelpers.bc` to `${BUILD_DIR}/{debug,release}/`.

Reply:
I agree with this

Reply:
Thanks.
I'll merge this after I fix the Java build failure.